lib/converters/v1.0.0/converter-v1-to-v2.js

```diff
@@ -184,43 +230,60 @@ _.assign(Builders.prototype, {
                 raw: 'raw',
                 binary: 'file'
             },
-            mode,
             data = {},
+            rawModeData,
+            dataMode = modes[requestV1.dataMode],
             retainEmpty = this.options.retainEmptyValues;
 
-        mode = modes[requestV1.dataMode];
-        !mode && (mode = 'raw');
-
-        if (mode === 'raw') {
-            data.mode = mode; // don't try to "clean" this up by moving it outside the "if". You'll break a test.
+        // set null body if neither data|rawModeData nor dataMode is set.
+        if (!dataMode && _.isEmpty(requestV1.data) && _.isEmpty(requestV1.rawModeData)) {
+            return retainEmpty ? null : undefined;
+        }
 
-            // If request.rawModeData exists, use it, or use request.data
-            data[mode] = _.isString(requestV1.rawModeData || requestV1.data) ?
-                requestV1.rawModeData || requestV1.data : '';
+        // set `rawModeData` if its a string
+        if (util.isString(requestV1.rawModeData)) {
```

Comment:
Need a comparative benchmark against `_.isString`. Add relevant comments explaining the need for this helper.

Reply:
<img width="968" alt="screenshot 2018-10-30 at 4 22 29 pm" src="https://user-images.githubusercontent.com/4865836/47713321-0abe1280-dc60-11e8-8efc-95a799d5815a.png">

Reply:
Switch to `_.isString`, there is a negligible performance impact.